### DISCUSSION TITLE
Custom icons/shapes in React native Calendars

### DISCUSSION THREAD

**Discussion by Ashutosh-107** [community/community#138622] [Programming Help]
### Body

I want to use this icon over a selected date in react native calendars.
![icons8-unavailable-48](https://github.com/user-attachments/assets/7ac814b1-b15e-4af1-a1a6-c3abc557f872)
How to do it?
How to add this type of shapes using custom css over a date in RN calendars?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by tpmst** [comment]
Hello @Ashutosh-107,

This setup will involve installing the necessary packages, configuring the calendar with a custom day component, and using an icon from react-native-vector-icons as a visual marker.

## Install Necessary Packages
First, you’ll need to add the calendar component and the vector icons package to your React Native project.

Install react-native-calendars:
```
npm install react-native-calendars
# or
yarn add react-native-calendars
```
Install react-native-vector-icons:

```
npm install react-native-vector-icons
# or
yarn add react-native-vector-icons
```

## Configure the Calendar with a Custom Day Component
You’ll create a custom day component that can show an icon on specific dates. This example will use the ban icon from FontAwesome, which resembles a block sign.

Create the Custom Day Component:
```
import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome'; // Import the icon component

const CustomDay = ({ date, state, marking }) => {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text style={{ textAlign: 'center' }}>{date.day}</Text>
      {marking.marked && <Icon name="block" size={20} color="#ff0000" />}
    </View>
  );
};
```
## Integrate the Calendar in Your Component
Now, integrate the react-native-calendars calendar with the custom day component in your main component.
```
import { Calendar } from 'react-native-calendars';

const MyCalendar = () => {
  return (
    <Calendar
      dayComponent={({ date, state, marking }) => <CustomDay date={date} state={state} marking={marking} />}
      markedDates={{
        '2024-08-15': { marked: true }, // Example date to show the icon
      }}
    />
  );
};

export default MyCalendar;
```

## Explanation:

- Custom Day Component: This component receives date, state, and marking props. The marking prop is used to determine if the day should display the icon.
- Marked Dates: This is where you specify which dates should display the icon. You can dynamically set these based on your application's needs.
- Icon Component: The Icon displays only if the marked property of a date is true. Adjust the icon’s name, size, and color according to your UI design.
